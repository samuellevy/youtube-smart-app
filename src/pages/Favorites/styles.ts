import styled from 'styled-components';
import { colors } from '../../global';

interface IContainer {
  active?: boolean;
  activeModule: number | 0;
}

export const Container = styled.div<IContainer>`
  padding: 25px 0 0 25px;
  padding-left: 250px;
  transform: ${(props) => (props.active ? `translateX(-140px) translateY(calc(-80vh*${props.activeModule}))` : `translateX(0) translateY(calc(-80vh*${props.activeModule}))`)};
  opacity: 1;
  transition: all 200ms linear;
`;
export const PageTitle = styled.h1`
  color: ${colors.gray};
  font-size: 1vw;
  text-transform: uppercase;
`;
export const Track = styled.div`
  display: flex;
`;
export const CategoryTitle = styled.h1`
  width: 100%;
`;
