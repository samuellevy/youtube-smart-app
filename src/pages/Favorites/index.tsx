/* eslint-disable array-callback-return */
import React, { useEffect, useState } from 'react';
import { useHistory } from 'react-router-dom';
import Track from '../../components/Track';
import { useFavoritesContext } from '../../context/FavoriteContext';
import { useKeyboardContext, changeComponent } from '../../context/KeyboardContext';
import { IVideo } from '../../dtos/IVideo';
import { IVideoCategory } from '../../dtos/IVideoCategory';
import { IVideoEntityReq } from '../../dtos/IVideoEntityReq';
import { api } from '../../services/api';

import * as S from './styles';

const Favorites: React.FC = () => {
  const { keyboard, dispatch }: any = useKeyboardContext();
  const { favorites }: any = useFavoritesContext();
  const [activePage, setActivePage] = useState(false);
  const [activeItem, setActiveItem] = useState(0);
  const [trackLoading, setTrackLoading] = useState(true);
  const history = useHistory();

  const [videoCategories, setVideoCategories] = useState<IVideoCategory[]>([] as IVideoCategory[]);

  useEffect(() => {
    dispatch(changeComponent('menu'));
    getCategories();
  }, []);

  useEffect(() => {
    if (keyboard.component === 'favorites') {
      setActivePage(true);
      controlHandler(keyboard.key);
    }
    if (keyboard.component === 'menu') {
      setActivePage(false);
    }
  }, [keyboard]);

  const controlHandler = (key: string) => {
    const newActiveItem = activeItem;
    switch (key) {
      case 'ArrowLeft':
        break;
      default:
        break;
    }
    setActiveItem(newActiveItem);
  };

  const handleOut = () => {
    dispatch(changeComponent('menu'));
  };

  const getCategories = async () => {
    const tmpVideoCategories: IVideoCategory[] = [{ id: 'favorites', title: 'Favoritos', videos: [] }];

    setVideoCategories(tmpVideoCategories);
  };

  const handlePlay = (id:string) => {
    history.push(`/video/?${id}`);
  };

  return (
    <S.Container active={activePage} activeModule={activeItem > 0 ? activeItem : 0}>
      {videoCategories.map((item: IVideoCategory, key) => (
        <Track
          key={item.id}
          active={key === activeItem}
          title={item.title}
          id={item.id}
          item={item}
          handleOut={() => { handleOut(); }}
          handlePlay={(id: string) => { handlePlay(id); }}
        />
      ))}
    </S.Container>
  );
};
export default Favorites;
