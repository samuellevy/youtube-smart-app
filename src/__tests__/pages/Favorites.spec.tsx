import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import Favorites from '../../pages/Favorites';

Enzyme.configure({ adapter: new Adapter() });

describe('<Favorites /> component', () => {
  it('should render properly', () => {
    const tree = renderer.create(<Favorites />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
