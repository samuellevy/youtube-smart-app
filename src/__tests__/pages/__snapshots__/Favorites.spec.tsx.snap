// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Favorites /> component should render properly 1`] = `
<div
  className="sc-jrAGrp fiSZBI"
/>
`;
